docs/book/user-guide/tutorials/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,590 @@
+
+if templates:
+    # Use the first matching template
+    template = templates[0]
+    print(f"Using template: {template.name} (ID: {template.id})")
+    
+    # Get the template's configuration
+    config = template.config_template
+    
+    # Update the configuration with step parameters
+    # Note: Parameters must be set at the step level rather than pipeline level
+    config["steps"] = {
+        "load_data": {
+            "parameters": {
+                "data_url": "s3://test-bucket/latest-data.csv",
+            }
+        },
+        "train_model": {
+            "parameters": {
+                "model_type": "gradient_boosting",
+            }
+        }
+    }
+    
+    # Trigger the pipeline with the updated configuration
+    run = Client().trigger_pipeline(
+        template_id=template.id,
+        run_configuration=config,
+    )
+    
+    print(f"Triggered pipeline run with ID: {run.id}")
+```
+
+#### Using the REST API:
+
+The REST API is ideal for external system integration, allowing you to trigger pipelines from non-Python environments:
+
+```bash
+# Step 1: Get the pipeline ID
+curl -X 'GET' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/pipelines?name=training_pipeline' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>'
+
+# Step 2: Get the template ID using the pipeline_id
+curl -X 'GET' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/run_templates?pipeline_id=<PIPELINE_ID>' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>'
+
+# Step 3: Trigger the pipeline with custom parameters
+curl -X 'POST' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/run_templates/<TEMPLATE_ID>/runs' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>' \
+  -d '{
+    "steps": {
+      "load_data": {
+        "parameters": {
+          "data_url": "s3://production-bucket/latest-data.csv"
+        }
+      },
+      "train_model": {
+        "parameters": {
+          "model_type": "gradient_boosting"
+        }
+      }
+    }
+  }'
+```
+
+> Note: When using the REST API, you need to specify parameters at the step level, not at the pipeline level. This matches how parameters are configured in the Python client.
+
+### Security Considerations for API Tokens
+
+When using the REST API for external systems, proper token management is critical:
+
+```python
+from zenml.client import Client
+
+# Create a service account for automated triggers
+service_account = Client().create_service_account(
+    name="pipeline-trigger-service",
+    description="Service account for external pipeline triggering"
+)
+
+# Generate API token with appropriate permissions
+token = Client().create_service_account_token(
+    service_account.id,
+    name="production-trigger-token",
+    description="Token for production pipeline triggers"
+)
+
+print(f"Store this token securely: {token.token}")
+# Make sure to save this token value securely
```

Comment:
You can also link here https://docs.zenml.io/api-reference/oss-api/getting-started#using-a-service-account-and-an-api-key